### ISSUE TITLE
More functionalities to VerifyReachability test

### ISSUE THREAD

**Issue by carl-baillargeon** [arista-netdevops-community/anta#341]
`VerifyReachability` test should be able to take a VRF as an input parameter and potentially other parameters.

This will improve the flexibility of the test.

**Comment by carl-baillargeon** [comment]
#315 is needed for this.

**Comment by JulioPDX** [comment]
Not sure if this applies, when testing with two different reachability tests, only the latest will apply vs both running. The same seems to apply when testing BGP neighbors in different VRFs.

Example test

```yaml
anta.tests.connectivity:
  - VerifyReachability:
      template_params:
        - dst: 8.8.8.8
          src: management0
anta.tests.connectivity:
  - VerifyReachability:
      template_params:
        - dst: 10.10.10.10
          src: management0
```

Results

```shell
╭────────────────────── Settings ──────────────────────╮
│ Running ANTA tests:                                  │
│ - ANTA Inventory contains 6 devices (AsyncEOSDevice) │
│ - Tests catalog contains 1 tests                     │
╰──────────────────────────────────────────────────────╯

[13:12:32] INFO     Running ANTA tests...                                                                             runner.py:94
  • Running NRFU Tests...100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6/6 • 0:00:02 • 0:00:00

                                                        All tests results                                                         
┏━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Device ┃ Test Name          ┃ Test Status ┃ Message(s)                      ┃ Test description                 ┃ Test category ┃
┡━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ spine1 │ VerifyReachability │ failure     │ Connectivity test failed for    │ Test the network reachability to │ connectivity  │
│        │                    │             │ the following                   │ one or many destination IP(s).   │               │
│        │                    │             │ source-destination pairs:       │                                  │               │
│        │                    │             │ [('management0',                │                                  │               │
│        │                    │             │ '10.10.10.10')]                 │                                  │               │
│ spine2 │ VerifyReachability │ failure     │ Connectivity test failed for    │ Test the network reachability to │ connectivity  │
│        │                    │             │ the following                   │ one or many destination IP(s).   │               │
│        │                    │             │ source-destination pairs:       │                                  │               │
│        │                    │             │ [('management0',                │                                  │               │
│        │                    │             │ '10.10.10.10')]                 │                                  │               │
│ leaf1  │ VerifyReachability │ failure     │ Connectivity test failed for    │ Test the network reachability to │ connectivity  │
│        │                    │             │ the following                   │ one or many destination IP(s).   │               │
│        │                    │             │ source-destination pairs:       │                                  │               │
│        │                    │             │ [('management0',                │                                  │               │
│        │                    │             │ '10.10.10.10')]                 │                                  │               │
│ leaf2  │ VerifyReachability │ failure     │ Connectivity test failed for    │ Test the network reachability to │ connectivity  │
│        │                    │             │ the following                   │ one or many destination IP(s).   │               │
│        │                    │             │ source-destination pairs:       │                                  │               │
│        │                    │             │ [('management0',                │                                  │               │
│        │                    │             │ '10.10.10.10')]                 │                                  │               │
│ leaf3  │ VerifyReachability │ failure     │ Connectivity test failed for    │ Test the network reachability to │ connectivity  │
│        │                    │             │ the following                   │ one or many destination IP(s).   │               │
│        │                    │             │ source-destination pairs:       │                                  │               │
│        │                    │             │ [('management0',                │                                  │               │
│        │                    │             │ '10.10.10.10')]                 │                                  │               │
│ leaf4  │ VerifyReachability │ failure     │ Connectivity test failed for    │ Test the network reachability to │ connectivity  │
│        │                    │             │ the following                   │ one or many destination IP(s).   │               │
│        │                    │             │ source-destination pairs:       │                                  │               │
│        │                    │             │ [('management0',                │                                  │               │
│        │                    │             │ '10.10.10.10')]                 │                                  │               │
└────────┴────────────────────┴─────────────┴─────────────────────────────────┴──────────────────────────────────┴───────────────┘
anta --enable nrfu -c anta-test.yml table  0.50s user 0.06s system 21% cpu 2.607 total
```

**Comment by carl-baillargeon** [comment]
@JulioPDX To run the same test multiple times, you need to put them under the same Python module:

```
anta.tests.connectivity:
  - VerifyReachability:
      template_params:
        - dst: 8.8.8.8
          src: management0
  - VerifyReachability:
      template_params:
        - dst: 10.10.10.10
          src: management0
```
You can also customize the test description with `result_overwrite` to differentiate your tests in your results:

```
anta.tests.connectivity:
  - VerifyReachability:
      template_params:
        - dst: 8.8.8.8
          src: management0
      result_overwrite:
        description: "Ping to Google DNS"
  - VerifyReachability:
      template_params:
        - dst: 10.10.10.10
          src: management0
      result_overwrite:
        description: "Ping to service A"
```
https://www.anta.ninja/v0.7.2/usage-inventory-catalog/#customize-test-description-and-categories

The behavior you experimented is expected with how the catalog is parsed in ANTA but it could be a good idea to add a note somewhere in the documentation. 

Thanks

**Comment by carl-baillargeon** [comment]
Resolved in #315
